source/gui/settingsDialogs.py

```diff
@@ -1636,31 +1636,53 @@ def makeSettings(self, settingsSizer):
 		settingsSizer.Add(sizer)
 
 		# Translators: The label for the edit field in symbol pronunciation dialog to change the pronunciation of a symbol.
-		changeSizer = wx.StaticBoxSizer(wx.StaticBox(self, wx.ID_ANY, _("Change symbol")), wx.VERTICAL)
+		changeSizer = wx.StaticBoxSizer(wx.StaticBox(self, wx.ID_ANY, _("Change selected symbol")), wx.VERTICAL)
+
+		# Used to ensure that event handlers call Skip(). In the case of handling EVT_KILL_FOCUS, not calling skip can 
+		# cause focus problems for controls. More generally the advice on the wx documentation is: "In general, it is 
+		# recommended to skip all non-command events to allow the default handling to take place. The command events are,
+		# however, normally not skipped as usually a single command such as a button click or menu item selection must 
+		# only be processed by one handler."
+		def SkipEventAndCall(some_func):
+			def WrapWithEventSkip(event):
+				if event:
+					event.Skip()
+				return some_func()
+			return WrapWithEventSkip
+
 		sizer = wx.BoxSizer(wx.HORIZONTAL)
+		# Translators: The label for the edit field in symbol pronunciation dialog to change the replacement text of a symbol.
 		sizer.Add(wx.StaticText(self, wx.ID_ANY, _("&Replacement")))
 		self.replacementEdit = wx.TextCtrl(self, wx.ID_ANY)
-		self.replacementEdit.Bind(wx.EVT_KILL_FOCUS, self.onSymbolEdited)
+		self.replacementEdit.Bind(wx.EVT_TEXT, SkipEventAndCall(self.onSymbolEdited))
 		sizer.Add(self.replacementEdit)
 		changeSizer.Add(sizer)
+
 		sizer = wx.BoxSizer(wx.HORIZONTAL)
 		# Translators: The label for the combo box in symbol pronunciation dialog to change the speech level of a symbol.
 		sizer.Add(wx.StaticText(self, wx.ID_ANY, _("&Level")))
 		symbolLevelLabels = characterProcessing.SPEECH_SYMBOL_LEVEL_LABELS
 		self.levelList = wx.Choice(self, wx.ID_ANY,choices=[
 			symbolLevelLabels[level] for level in characterProcessing.SPEECH_SYMBOL_LEVELS])
-		self.levelList.Bind(wx.EVT_KILL_FOCUS, self.onSymbolEdited)
+		self.levelList.Bind(wx.EVT_CHOICE, SkipEventAndCall(self.onSymbolEdited))
 		sizer.Add(self.levelList)
 		changeSizer.Add(sizer)
+
 		sizer = wx.BoxSizer(wx.HORIZONTAL)
 		# Translators: The label for the combo box in symbol pronunciation dialog to change when a symbol is sent to the synthesizer.
 		sizer.Add(wx.StaticText(self, wx.ID_ANY, _("&Send actual symbol to synthesizer")))
 		symbolPreserveLabels = characterProcessing.SPEECH_SYMBOL_PRESERVE_LABELS
 		self.preserveList = wx.Choice(self, wx.ID_ANY,choices=[
 			symbolPreserveLabels[mode] for mode in characterProcessing.SPEECH_SYMBOL_PRESERVES])
-		self.preserveList.Bind(wx.EVT_KILL_FOCUS, self.onSymbolEdited)
+		self.preserveList.Bind(wx.EVT_CHOICE, SkipEventAndCall(self.onSymbolEdited))
 		sizer.Add(self.preserveList)
 		changeSizer.Add(sizer)
+
+		# disable the "change symbol" controls until a valid item is selected.

```

Comment:
While that is a simpler fix, I think its at the cost of the UX. I touched on this in the description of this PR, but perhaps did not explain well enough why I chose to do it this way. From a visual standpoint, because the fields are disabled, one is directed to interact with the list first. There is nothing else in the dialog to interact with. Then when an item is selected or added, the fields become available. This helps to reinforce that the fields are for the item selected / just added to the list.